### PR TITLE
gracefully fail with unsupported HTTP methods

### PR DESCRIPTION
**Pull request by tangiel**

### CODE REVIEW

File: endpoints-framework/src/main/java/com/google/api/server/spi/dispatcher/PathDispatcher.java [hunk 1]
```diff
@@ -42,12 +42,14 @@ private PathDispatcher(Builder<ContextT> builder) {
   public boolean dispatch(String httpMethod, String path, ContextT context) throws IOException {
     Preconditions.checkNotNull(httpMethod, "httpMethod");
     Preconditions.checkNotNull(path, "path");
-    Result<DispatcherHandler<ContextT>> result =
-        trie.resolve(HttpMethod.valueOf(httpMethod.toUpperCase()), path);
-    if (result != null) {
-      context.setRawPathParameters(result.getRawParameters());
-      result.getResult().handle(context);
-      return true;
+    HttpMethod method = HttpMethod.fromString(httpMethod.toUpperCase());
```
Comment (inklesspen) [inline endpoints-framework/src/main/java/com/google/api/server/spi/dispatcher/PathDispatcher.java]: Why not put the `toUpperCase` inside the `fromString` method, so it doesn't matter what case the method is in?

File: endpoints-framework/src/test/java/com/google/api/server/spi/dispatcher/PathDispatcherTest.java [hunk 2]
```diff
@@ -121,4 +121,10 @@ public void builderNullHandler() {
       // expected
     }
   }
+
+  @Test
+  public void methodNotKnown() throws IOException {
+    PathDispatcher<DispatcherContext> dispatcher = PathDispatcher.builder().build();
+    assertThat(dispatcher.dispatch("HEAD", "test/one/two/3", context)).isFalse();
```
Comment (inklesspen) [inline endpoints-framework/src/test/java/com/google/api/server/spi/dispatcher/PathDispatcherTest.java]: isn't `HEAD` supposed to be legal on any url for which `GET` is legal? the RFC suggests that's the case (it uses SHOULD, though, not MUST)

perhaps a made-up method would be better here

File: endpoints-framework/src/main/java/com/google/api/server/spi/dispatcher/PathDispatcher.java (hunk 1 above)
Comment (tangiel) [inline endpoints-framework/src/main/java/com/google/api/server/spi/dispatcher/PathDispatcher.java]: I did. This is redundant. Oops!

File: endpoints-framework/src/test/java/com/google/api/server/spi/dispatcher/PathDispatcherTest.java (hunk 2 above)
Comment (tangiel) [inline endpoints-framework/src/test/java/com/google/api/server/spi/dispatcher/PathDispatcherTest.java]: Done.

File: endpoints-framework/src/main/java/com/google/api/server/spi/dispatcher/PathDispatcher.java (hunk 1 above)
Comment (inklesspen) [inline endpoints-framework/src/main/java/com/google/api/server/spi/dispatcher/PathDispatcher.java]: heh, so you did.

**Pull request merged**